pkg/controller/utils/pod/create_test.go

```diff
@@ -129,3 +133,34 @@ func Test_overwriteResourcesFromNode(t *testing.T) {
 		})
 	}
 }
+
+func Test_overwriteResourcesFromEdsNode(t *testing.T) {
+	templateOriginal := &corev1.PodTemplateSpec{
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{Name: "container1"},
+			},
+		},
+	}
+
+	templateCopy := templateOriginal.DeepCopy()
+	// nil case, no change to template
+	edsNode := &datadoghqv1alpha1.ExtendedDaemonsetSetting{}
+	overwriteResourcesFromEdsNode(templateOriginal, edsNode)
+	assert.Equal(t, templateCopy, templateOriginal)
+
+	resouresRef := corev1.ResourceList{
+		"cpu":    resource.MustParse("0.1"),
+		"memory": resource.MustParse("20M"),
+	}
+	edsNode = datadoghqv1alpha1test.NewExtendedDaemonsetSetting("foo", "bar", "reference", &datadoghqv1alpha1test.NewExtendedDaemonsetSettingOptions{
+		CreationTime: time.Now(),
+		Resources: map[string]corev1.ResourceRequirements{
+			"container1": {
+				Requests: resouresRef,
+			},
+		},
+	})
+	overwriteResourcesFromEdsNode(templateOriginal, edsNode)
+	assert.NotEqual(t, templateCopy, templateOriginal)
```

Comment:
Apart from checking that they're not equal, is it worth testing that what changed is really the changes expected?